### PR TITLE
SOANRJAVA-1062 Removes usage of ast node in SynchronizedClassUsageCheck

### PR DESCRIPTION
**Pull request by m-g-sonar**
Note that class `AstNodeTokensMatcher.java` have been removed has it is now useless.

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

File: java-checks/src/main/java/org/sonar/java/checks/SynchronizedClassUsageCheck.java [hunk 1]
```diff
@@ -58,88 +59,83 @@
...
-    scan(tree.typeParameters());
-    scan(tree.returnType());
-    scan(tree.defaultValue());
-    scan(tree.block());
-    if (!isOverriding(tree)) {
-      for (VariableTree param : tree.parameters()) {
-        reportIssueIfDeprecatedType(param.type());
-      }
-      reportIssueIfDeprecatedType(tree.returnType());
+    DeprecatedTypeVisitor visitor = new DeprecatedTypeVisitor(this);
```
Comment (merciesa) [inline java-checks/src/main/java/org/sonar/java/checks/SynchronizedClassUsageCheck.java]: Why not simply tree.accept(new DeprecatedTypeVisitor(this)); ? :)

File: java-checks/src/main/java/org/sonar/java/checks/SynchronizedClassUsageCheck.java [hunk 2]
```diff
@@ -58,88 +59,83 @@
...
-    }
+  private static class DeprecatedTypeVisitor extends BaseTreeVisitor {
 
-  }
+    private final IssuableSubscriptionVisitor check;
 
-  private boolean reportIssueIfDeprecatedType(@Nullable ExpressionTree tree) {
-    if (tree == null) {
-      return false;
+    public DeprecatedTypeVisitor(IssuableSubscriptionVisitor check) {
```
Comment (merciesa) [inline java-checks/src/main/java/org/sonar/java/checks/SynchronizedClassUsageCheck.java]: You can remove the constructor and 'check' field by removing the 'static' modifier on the class and invoking addIssue directly (without prefixing it with check.) in reportIssueOnDeprecatedType.

**Comment by merciesa** [comment]
Two comments left to your appreciation. Otherwise LGTM

**Pull request closed without merging**